K8s docs v2 (#643)
ngrok/ngrok-docs: docs/getting-started/kubernetes.mdx

import K8sIngressInstallExample from "/examples/k8s/install-ingress.mdx";
import K8sGatewayAPIInstallExample from "/examples/k8s/install-gateway-api.mdx";

import K8sIngressModifyHeadersExample from "/examples/k8s/http-request-headers.mdx";
import K8sGatewayModifyHeadersExample from "/examples/k8s/http-request-headers-gateway.mdx";

import K8sIngressRedirectExample from "/examples/k8s/http-request-redirect.mdx";
import K8sGatewayRedirectExample from "/examples/k8s/http-request-redirect-gateway.mdx";

import StaticDomainKubernetesExample from "/examples/k8s/http-static-domain.mdx";
import StaticDomainKubernetesGatewayExample from "/examples/k8s/http-static-domain-gateway.mdx";

import StaticDomainKubernetesPathExample from "/examples/k8s/http-static-domain-path.mdx";
import StaticDomainKubernetesGatewayPathExample from "/examples/k8s/http-static-domain-gateway-path.mdx";

# Quickstart: Kubernetes Operator

This quickstart will use the [ngrok Kubernetes Operator](https://github.com/ngrok/kubernetes-ingress-controller) to put an application running on a Kubernetes Cluster on the internet and add some basic policies.

## What is the ngrok Kubernetes Operator?

> The ngrok Kubernetes Operator is the best way to use ngrok if your applications run as Services in a Kubernetes cluster. Some of the benefits of the operator are:
> 
> - The Operator can be easily installed on a cluster via Helm and is compatible with the most recent versions of Kubernetes.
> - The Operator supports both [Ingress CRDs](https://kubernetes.io/docs/concepts/services-networking/ingress/) as well as the new [Gateway API](https://gateway-api.sigs.k8s.io/) using the same installation.
> - When deployed to a cluster, the Operator allows developers to self-service public endpoints to their apps and services using a shared ngrok account.
> - The Operator manages the creation of ngrok resources, leaving developers to focus on traffic and routing patterns.
> - The Operator supports the configuration of resources like custom domains, modules, and policies.
> - The Operator supports all the cloud benefits of ngrok, including our global points of presense, GSLB, and traffic inspector.

Reviewer: ```suggestion
The ngrok Kubernetes Operator is the best way to use ngrok if your applications run as services in a Kubernetes cluster. Key benefits of the operator include:

- **Easy Installation:** The Operator can be easily installed on a cluster via Helm and is compatible with the latest versions of Kubernetes.
- **Broad Support:** The Operator supports both [Ingress CRDs](https://kubernetes.io/docs/concepts/services-networking/ingress/) and the new [Gateway API](https://gateway-api.sigs.k8s.io/) using the same installation.
- **Developer Self-Service:** Once deployed, the Operator enables developers to self-service public endpoints for their apps and services using a shared ngrok account.
- **Resource Management:** The Operator manages the creation of ngrok resources, allowing developers to focus on traffic and routing patterns.
- **Advanced Configuration:** The Operator supports the configuration of resources like custom domains, modules, and policies.
- **Cloud Benefits:** The Operator supports all the cloud benefits of ngrok, including our global points of presence, GSLB, and traffic inspector.
```

## Step 1: Install Kubernetes Operator

Installing the ngrok Operator is easy using Helm.

<Tabs groupId="k8s" queryString="k8s-install">
	<TabItem value="ingress" label="Ingress Controller" default>
		<K8sIngressInstallExample />
	</TabItem>
	<TabItem value="gatewayAPI" label="Gateway API">
		<K8sGatewayAPIInstallExample />
	</TabItem>
</Tabs>

> <br />
> 

Reviewer: ```suggestion
```

## Step 2: Connect your service to an Edge

This would create an Edge on with the Domain `example.ngrok.app` that routes traffic to the Kubernetes service `example-service` running on port `80`.

<Tabs groupId="k8s" queryString="k8s-basic">
	<TabItem value="ingress" label="Ingress Controller" default>
		<StaticDomainKubernetesExample />
	</TabItem>
	<TabItem value="gatewayAPI" label="Gateway API">
		<StaticDomainKubernetesGatewayExample />
	</TabItem>
</Tabs>

> ## Step 3: Route Traffic based on a path

Reviewer 2: are we capitalizing the titles or not? seems like a mix throughout the PR. can we standardize?

<Tabs groupId="k8s" queryString="k8s-basic-path">
	<TabItem value="ingress" label="Ingress Controller" default>
		<StaticDomainKubernetesPathExample />
	</TabItem>
	<TabItem value="gatewayAPI" label="Gateway API">
		<StaticDomainKubernetesGatewayPathExample />
	</TabItem>
</Tabs>

## Step 4: Modify Headers for incoming requests

<Tabs groupId="k8s" queryString="k8s-request-headers">
	<TabItem value="ingress" label="Ingress Controller" default>
		<K8sIngressModifyHeadersExample />
	</TabItem>
	<TabItem value="gatewayAPI" label="Gateway API">
		<K8sGatewayModifyHeadersExample />
	</TabItem>
</Tabs>

## Step 5: Redirect Traffic

<Tabs groupId="k8s" queryString="k8s-request-redirect">
	<TabItem value="ingress" label="Ingress Controller" default>
		<K8sIngressRedirectExample />
	</TabItem>
	<TabItem value="gatewayAPI" label="Gateway API">
		<K8sGatewayRedirectExample />
	</TabItem>
</Tabs>

## Next Steps

- Star the [Kubernetes Operator GitHub repo](https://github.com/ngrok/kubernetes-ingress-controller) and follow
  it for updates.

> - Browse [Go examples of HTTP Endpoints](/http/?cty=k8s) of other ways ngrok can augment your application.

Reviewer 2: ```suggestion
- Browse our [examples of HTTP Endpoints](/http/?cty=k8s) to see other ways ngrok can augment your application.
```

import K8sIngressInstallExample from "/examples/k8s/install-ingress.mdx";
import K8sGatewayAPIInstallExample from "/examples/k8s/install-gateway-api.mdx";

import K8sIngressModifyHeadersExample from "/examples/k8s/http-request-headers.mdx";
import K8sGatewayModifyHeadersExample from "/examples/k8s/http-request-headers-gateway.mdx";

import K8sIngressRedirectExample from "/examples/k8s/http-request-redirect.mdx";
import K8sGatewayRedirectExample from "/examples/k8s/http-request-redirect-gateway.mdx";

import StaticDomainKubernetesExample from "/examples/k8s/http-static-domain.mdx";
import StaticDomainKubernetesGatewayExample from "/examples/k8s/http-static-domain-gateway.mdx";

import StaticDomainKubernetesPathExample from "/examples/k8s/http-static-domain-path.mdx";
import StaticDomainKubernetesGatewayPathExample from "/examples/k8s/http-static-domain-gateway-path.mdx";

# Quickstart: Kubernetes Operator

This quickstart will use the [ngrok Kubernetes Operator](https://github.com/ngrok/kubernetes-ingress-controller) to put an application running on a Kubernetes Cluster on the internet and add some basic policies

## What is the ngrok Kubernetes Operator?

> The ngrok Kubernetes Operator is the best way to use ngrok if your applications run as Services in a Kubernetes cluster. It supports both [Ingress CRDs](https://kubernetes.io/docs/concepts/services-networking/ingress/) as well as the new [Gateway API](https://gateway-api.sigs.k8s.io/). The ngrok Kubernetes Operator is deployed to a cluster and allows developers to self-service public endpoints to their apps and services using a shared ngrok account. The operator watches for Kubernetes objects and creates the corresponding ngrok endpoints and configurations.

Reviewer 2: i think we should expand this paragraph a bit and include some of the benefits of using the operator.

## Step 1: Install Kubernetes Operator

Installing the ngrok Operator is easy using Helm.

<Tabs groupId="k8s" queryString="k8s-install">
	<TabItem value="ingress" label="Ingress Controller" default>
		<K8sIngressInstallExample />
	</TabItem>
	<TabItem value="gatewayAPI" label="Gateway API">
		<K8sGatewayAPIInstallExample />
	</TabItem>
</Tabs>

<br />

## Step 2: Connect your service to an Edge

> This would create an Edge on with the Domain `example.ngrok.app` that routes traffic to the Kubernetes service `example-service` running on port 80.

Reviewer: ```suggestion
This would create an Edge on with the Domain `example.ngrok.app` that routes traffic to the Kubernetes service `example-service` running on port `80`.
```

<Tabs groupId="k8s" queryString="k8s-basic">
	<TabItem value="ingress" label="Ingress Controller" default>
		<StaticDomainKubernetesExample />
	</TabItem>
	<TabItem value="gatewayAPI" label="Gateway API">
		<StaticDomainKubernetesGatewayExample />
	</TabItem>
</Tabs>

## Step 3: Route Traffic based on a path

> This would create an Edge on with the Domain `example.ngrok.app` that routes traffic from example.ngrok.app/foo to the Kubernetes service `example-service` running on port 80.

Reviewer: ```suggestion
This would create an Edge on with the Domain `example.ngrok.app` that routes traffic from `example.ngrok.app/foo` to the Kubernetes service `example-service` running on port `80`.
```
